Geometry/CaloEventSetup/plugins/CaloTowerConstituentsMapBuilder.cc

```diff
@@ -130,17 +130,16 @@ void CaloTowerConstituentsMapBuilder::assignEEtoHE(const CaloGeometry* geometry,
   //get list of EE detids
   const std::vector<DetId>& vec(geomEE->getValidDetIds());
   //loop over EE detids
-  for(std::vector<DetId>::const_iterator detId_itr =vec.begin();
-      detId_itr != vec.end(); detId_itr++){
+  for(auto detId_itr : vec){
```

Comment:
It should be as long as the class used the compiler generated default constructors.